test/topology/test_mutation_schema_change.py

```diff
@@ -0,0 +1,157 @@
+    await wait_for(server_c_sees_another_server, time.time() + 45, period=.1)
+
+    logger.info("Driver connecting to C %s", server_c)
+    manager.ccluster = manager.con_gen([server_c.ip_addr], manager.port, manager.use_ssl)
+    manager.cql = manager.ccluster.connect()
+
+    stmt = f"UPDATE {t} "                        \
+           f"SET   {t.columns[3].name} = %s "  \
+           f"WHERE {t.columns[0].name} = %s "  \
+           f"IF    {t.columns[3].name} = %s"
+    query = SimpleStatement(stmt, consistency_level=ConsistencyLevel.ONE)
+    for seed in seeds:
+        logger.info("UPDATE with seed %s", seed)
+        await manager.cql.run_async(query, parameters=[t.columns[3].val(seed + 1), # v_01 = seed + 1
+                                                       t.columns[0].val(seed),     # pk = seed
+                                                       t.columns[3].val(seed)])    # v_01 == seed
+
+
+@pytest.mark.asyncio
+async def test_mutation_schema_change_restart(manager, random_tables):
+    """
+        Cluster A, B, C
+        create table
+        stop C
+        change schema + do lwt write + change schema
+        stop B
+        restart A
+        start C
+        do lwt write to the same key through A
+    """
+    server_a, server_b, server_c = await manager.running_servers()
+    t = await random_tables.add_table(ncolumns=5, pks=1)
+    manager.driver_close()
+    # Reduce the snapshot thresholds
+    await manager.mark_dirty()
+    errs = [inject_error_one_shot(manager.api, s.ip_addr, 'raft_server_snapshot_reduce_threshold')
+            for s in [server_a, server_b, server_c]]
+    await asyncio.gather(*errs)
+
+    logger.info("Stopping C %s", server_c)
+    await manager.server_stop_gracefully(server_c.server_id)
+    await manager.driver_connect()
+
+    await inject_error_one_shot(manager.api, server_a.ip_addr,
+                                'raft_server_reduce_threshold')
+    async with inject_error(manager.api, server_b.ip_addr, 'paxos_error_before_learn'):
+        await t.add_column()
+        ROWS = 1
+        seeds = [t.next_seq() for _ in range(ROWS)]
+        stmt = f"INSERT INTO {t} ({','.join(c.name for c in t.columns)}) " \
+               f"VALUES ({', '.join(['%s'] * len(t.columns))}) "           \
+               f"IF NOT EXISTS"
+        query = SimpleStatement(stmt, consistency_level=ConsistencyLevel.ONE)
+        for seed in seeds:
+            logger.info("INSERT row seed %s", seed)
+            await manager.cql.run_async(query, parameters=[c.val(seed) for c in t.columns])
+        await t.add_column()
+
+    manager.driver_close()
+
+    logger.info("Stopping B %s", server_b)
+    await manager.server_stop_gracefully(server_b.server_id)
+    logger.info("Restarting A %s", server_a)
+    await manager.server_restart(server_a.server_id)
+    logger.info("Starting C %s", server_c)
+    await manager.server_start(server_c.server_id)
+
+    async def server_c_sees_another_server():
```

Comment:
Shouldn't we wait for `A` to see another server here, because we'll want to write through `A`?
(not sure I understand why this wait is needed)

Reply:
It doesn't make any difference.

Reply:
Fixed in latest